### PR TITLE
Show geographies with expired protections

### LINKED ISSUE #67: Need to include geographies that have expired
We don't always know the expiration date, and currently places that might have expired (but might not have as the dates change frequently) are missing on the map (ie. Massachusetts). Perhaps  we could show these as cross hatched.

### LINKED ISSUE #102: Add expiration date to housing protection popup
So that viewers can see the expiration date

### PR DESCRIPTION
**Pull request by clhenrick**
Resolves #67 
Resolves #102

TODO:

- [x] pattern fills for counties, states, nations
- [x] pattern fills for cities, OR 
- [ ] separate marker icons for cities? (could use feedback on)
- [x] update legend to include explanation that pattern fills = expiring protections
- [x] fix issue with map container being scrollable (probably due to the addition of the hidden `<svg>` element.
- [x] add policy expiration dates to infowindow (see #102)

- ~~adds the [Leaflet Pattern Plugin](https://github.com/teastman/Leaflet.pattern) to visualize geographies with expired protections as a striped pattern fill.~~

☝️ for some reason the Leaflet Pattern Plugin doesn't seem to work 100% correctly when loaded using Webpack. Rather than debug it I'm using a work around with class names and CSS `fill: url(#pattern-id)` to apply the pattern fills.

### CODE REVIEW

**Comment by clhenrick** [comment]
It looks like the Leaflet Pattern Plugin doesn't seem to work correctly with Webpack. When I try to use it, it fails to add the necessary `<defs>` declaration in Leaflet's `<svg>` element, so the pattern fails to draw:

image

One work around that may be worth trying would be to add a hidden SVG element in the `index.html` page somewhere that contains the pattern definition, and then append the pattern fill after the map initially renders, but it seems a bit hacky. It might not work as it would have to re-render each time the map is zoomed or panned.

**Comment by clhenrick** [comment]
Using CSS class names I am able to apply the pattern fill without the use of the leaflet.pattern plugin:

image

Question for the group, do we also want to use this technique for cities? It technically works but the city circle markers start to get a bit lost, so maybe it would be better to use actual marker icons (cc @bretthalperin)

image

**Comment by clhenrick** [comment]
Updated the legend to include an explanation of what the stripes mean:

image

File: public/index.html [hunk 1]
```diff
@@ -223,5 +262,198 @@ <h2>Resources</h2>
         {{/resources}}
       </div>
     </script>
+    <!-- Leaflet map SVG patterns -->
```
Comment (timhitchins) [inline public/index.html:265]: Interesting to see how this actually works.

File: public/index.html [hunk 2]
```diff
@@ -223,5 +262,198 @@ <h2>Resources</h2>
...
+        <!-- city protections have a slightly modified pattern to better differentiate them -->
+        <pattern
+          id="pattern-few-protections--city-level"
+          x="0"
+          y="0"
+          width="8"
+          height="8"
+          patternUnits="userSpaceOnUse"
+          patternContentUnits="userSpaceOnUse"
+          patternTransform="rotate(135)"
```
Comment (timhitchins) [inline public/index.html:371]: Seems like a good fix and works visually imo.

File: src/lib/leaflet.js [hunk 3]
```diff
@@ -13,4 +13,5 @@ export const rentStrikeIcon = new L.Icon({
   className: "icon-rent-strike",
 });
 
+window.L = L;
```
Comment (timhitchins) [inline src/lib/leaflet.js:16]: Why was this needed?

File: src/styles/_base.scss [hunk 4]
```diff
@@ -29,6 +29,12 @@ a {
   width: 100vw;
 }
 
+#map-pattern-fills {
+  // the svg for the map's pattern fills should be hidden in the UI
```
Comment (timhitchins) [inline src/styles/_base.scss:33]: Interesting way to deal with this.

File: src/styles/mixins.scss [hunk 5]
```diff
@@ -0,0 +1,34 @@
+/*
```
Comment (timhitchins) [inline src/styles/mixins.scss:1]: Thank  you for the details.

File: src/utils/queries.js [hunk 6]
```diff
@@ -13,16 +13,19 @@ import {
 
 export const citiesCartoQuery = `
```
Comment (timhitchins) [inline src/utils/queries.js:14]: These updates look good to me.

File: src/lib/leaflet.js (hunk 3 above)
Comment (clhenrick) [reply src/lib/leaflet.js:16]: Ah that’s left over from when I was trying to debug the Leaflet Pattern plug-in. I’ll remove it before merging. Thanks for catching.

File: src/styles/_base.scss (hunk 4 above)
Comment (clhenrick) [reply src/styles/_base.scss:33]: Maybe I should move that block to the map sass partial

**Comment by clhenrick** [comment]
Thanks @timhitchins, waiting on @erinmcel and possibly some of the research folks to review the Netlify preview before merging.

**Comment by erinmcel** [comment]
Sorry for the delay. I think it looks great, and the popup does too!

File: public/index.html (hunk 2 above)
Comment (clhenrick) [reply public/index.html:371]: @timhitchins would you mind saying more about this?

**Comment by clhenrick** [comment]
Confirmed that this works on IE11 as well:
image
